ext/bcmath/libbcmath/src/str2num.c

```diff
@@ -76,6 +76,35 @@ static const char *bc_count_digits(const char *str, const char *end)
 	return str;
 }
 
+static inline const char *bc_skip_zero_reverse(const char *str, const char *end)
+{
+	/* Check in bulk */
+#ifdef __SSE2__
+	const __m128i c_zero_repeat = _mm_set1_epi8((signed char) '0');
+	while (str - sizeof(__m128i) >= end) {
+		str -= sizeof(__m128i);
+		__m128i bytes = _mm_loadu_si128((const __m128i *) str);
+		/* Checks if all numeric strings are equal to '0'. */
+		bytes = _mm_cmpeq_epi8(bytes, c_zero_repeat);
+
+		int mask = _mm_movemask_epi8(bytes);
+		/* The probability of having 16 trailing 0s in a row is very low, so we use EXPECTED. */
+		if (EXPECTED(mask != 0xffff)) {
+			/* Move the pointer back and check each character in loop. */
+			str += sizeof(__m128i);
+			break;
+		}
```

Comment:
I can also use code like the following, but a while loop has always been faster. This may be because the number of calculations increases by one.

```
return str + sizeof(__m128i) - __builtin_clz(~mask);
```